jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/MobileConnectRequest.kt

```diff
@@ -0,0 +1,9 @@
+package com.radixdlt.sargon.extensions
+
+import com.radixdlt.sargon.MobileConnectRequest
+import com.radixdlt.sargon.newMobileConnectRequest
+
+@Throws(SargonException::class)
+fun MobileConnectRequest.Companion.parseFrom(url: String) = runCatching {
```

Comment:
should probably not only accept `String` should probably have a method taking `Uri` too, update Rust Sargon accordingly if applicable?

Reply:
why? deep link comes in to android wallet as `android.net.Uri` so even different type then `java.net.URI`

Reply:
Ok.

We can also do a UniFFI newtype, that translates into `android.net.Uri` and use it here. That way you need not pass in a `String`.

But maybe not worth it.

Reply:
using android sdk classes in business logic mean that we need android SDK to run tests, we prefer to avoid it to have them easier and faster to run. If you really want I can pass `Uri` using `java.net.Uri`, converting deep link to string first and then to `URI`